src/plugins/discover/public/application/main/state_management/discover_state.ts

```diff
@@ -407,17 +412,24 @@ export function getDiscoverStateContainer({
     }
   };
 
+  const setIsLoading = (value: boolean) => {
+    internalStateContainer.transitions.setIsLoading(value);
+  };
+
   const onDataViewEdited = async (editedDataView: DataView) => {
+    setIsLoading(true);
+    const edited = editedDataView.id;
     if (editedDataView.isPersisted()) {
       // Clear the current data view from the cache and create a new instance
       // of it, ensuring we have a new object reference to trigger a re-render
-      services.dataViews.clearInstanceCache(editedDataView.id);
+      services.dataViews.clearInstanceCache(edited);
       setDataView(await services.dataViews.create(editedDataView.toSpec(), true));
     } else {
       await updateAdHocDataViewId();
     }
-    loadDataViewList();
+    await loadDataViewList();
```

Comment:
I first thought, ha! got you flaky test monster, a missing await, however it didn't help, but it should be changed anyway